src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -106,6 +105,17 @@ export const portfolio = createSlice({
       // TODO(gomes): do we also want to clear draftState.accountMetadata entries themselves?
       // Theoretically, not doing so would make reloading these easier?
     },
+    clearWalletPortfolioState: (draftState, { payload }: { payload: string }) => {
+      const walletId = payload
+
+      delete draftState.wallet.byId[walletId]
+      draftState.wallet.ids = draftState.wallet.ids.filter(id => id !== walletId)
+      delete draftState.accountMetadata.byId[walletId]
+      draftState.accountMetadata.ids = draftState.accountMetadata.ids.filter(id => id !== walletId)
+      delete draftState.accountBalances.byId[walletId]
+      draftState.accountBalances.ids = draftState.accountBalances.ids.filter(id => id !== walletId)
```

Comment:
Thought: these filters are O(n), but it's probably not worth the additional code complexity to get O(1) considering the low number of wallets/accounts we are dealing with.